llvm/lib/CodeGen/ExpandVectorPredication.cpp

```diff
@@ -465,6 +465,21 @@ CachingVPExpander::expandPredicationInMemoryIntrinsic(IRBuilder<> &Builder,
         VPI.getName());
     break;
   }
+  case Intrinsic::vp_load_ff: {
+    // Mask out all but the first lane.
+    Value *NewMask = ConstantInt::getFalse(MaskParam->getType());
+    NewMask = Builder.CreateInsertElement(
+        NewMask, ConstantInt::getTrue(MaskParam->getType()->getScalarType()),
+        (uint64_t)0);
```

Comment:
Why do we need the (uint64_t) cast?

Reply:
There are 2 CreateInsertElement functions. The other takes a Value*. Because 0 is implicitly convertible to nullptr it becomes ambiguous. Using uint64_t makes the type match exactly so implicit conversions don't get checked.